stock_dropshipping_dual_invoice/wizard/stock_invoice_onshipping.py

```diff
@@ -21,6 +23,23 @@
 class StockInvoiceOnshipping(models.TransientModel):
     _inherit = "stock.invoice.onshipping"
 
+    @api.model
+    def _get_journal_type(self):
+        res_ids = self.env.context.get('active_ids', [])
+        pickings = self.env['stock.picking'].browse(res_ids)
+        pick = pickings and pickings[0]
+        if pick.move_lines:
+            src_usage = pick.move_lines[0].location_id.usage
+            dest_usage = pick.move_lines[0].location_dest_id.usage
+            if src_usage == 'supplier' and dest_usage == 'customer':
+                moves = pick.move_lines.filtered('purchase_line_id')
+                moves = moves and moves[0]
```

Comment:
This is an annoying thing I have to introduce, because in some moment after the launch of the new API, they have added an ensure_one() in the field get method, so now we cannot access first record of the recordset with dot notation when there are more than one record.

With your expression, it will get a list of values, but it will be empty in case there is no records, so we have to test also if the list is empty.